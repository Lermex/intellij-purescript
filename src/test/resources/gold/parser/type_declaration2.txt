Purescript File
  PSASTWrapperElement(Program)
    PSASTWrapperElement(Module)
      PsiElement(module)('module')
      PsiElement(whitespace)(' ')
      PSASTWrapperElement(Qualified)
        PSASTWrapperElement(ModuleName)
          PsiElement(proper name)('Main')
      PsiElement(whitespace)(' ')
      PsiElement(where)('where')
      PsiElement(whitespace)('\n\n')
      PSASTWrapperElement(TypeSynonymDeclaration)
        PsiElement(type)('type')
        PsiElement(whitespace)(' ')
        PSASTWrapperElement(TypeConstructor)
          PsiElement(proper name)('Woo')
          PsiElement(whitespace)(' ')
        PSASTWrapperElement(GenericIdentifier)
          PsiElement(identifier)('a')
          PsiElement(whitespace)(' ')
        PsiElement(=)('=')
        PsiElement(whitespace)(' ')
        PSASTWrapperElement(Type)
          PSASTWrapperElement(TypeAtom)
            PSASTWrapperElement(ForAll)
              PsiElement(forall)('forall')
              PsiElement(whitespace)(' ')
              PSASTWrapperElement(GenericIdentifier)
                PsiElement(identifier)('a')
              PsiElement(.)('.')
              PsiElement(whitespace)(' ')
              PSASTWrapperElement(ConstrainedType)
                PSASTWrapperElement(Type)
                  PSASTWrapperElement(TypeAtom)
                    PSASTWrapperElement(ObjectType)
                      PsiElement({)('{')
                      PSASTWrapperElement(Row)
                        PSASTWrapperElement(LocalIdentifier)
                          PSASTWrapperElement(identifier)
                            PsiElement(identifier)('name')
                          PsiElement(whitespace)(' ')
                        PsiElement(::)('::')
                        PsiElement(whitespace)(' ')
                        PSASTWrapperElement(Type)
                          PSASTWrapperElement(TypeAtom)
                            PSASTWrapperElement(GenericIdentifier)
                              PsiElement(identifier)('a')
                      PsiElement(})('}')