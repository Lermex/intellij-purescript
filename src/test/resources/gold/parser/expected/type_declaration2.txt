Purescript File
  PSModuleDeclarationImpl(MODULE_DECLARATION)
    PsiElement(PSTokenType.module)('module')
    PSModuleNameImpl(MODULE_NAME)
      PsiElement(PSTokenType.PROPER_NAME)('Main')
    PsiElement(PSTokenType.where)('where')
  PSDeclarationsImpl(DECLARATIONS)
    PSDeclarationImpl(DECLARATION)
      PSTypeDeclarationImpl(TYPE_DECLARATION)
        PsiElement(PSTokenType.type)('type')
        PSDeclarationRefTypeListImpl(DECLARATION_REF_TYPE_LIST)
          PSDeclarationRefTypeImpl(DECLARATION_REF_TYPE)
            PsiElement(PSTokenType.PROPER_NAME)('Woo')
        PSDeclarationRefIdListImpl(DECLARATION_REF_ID_LIST)
          PSDeclarationRefIdImpl(DECLARATION_REF_ID)
            PsiElement(PSTokenType.ID)('a')
        PsiElement(PSTokenType.=)('=')
        PSTypeDeclarationItemImpl(TYPE_DECLARATION_ITEM)
          PSGenericModifierImpl(GENERIC_MODIFIER)
            PsiElement(PSTokenType.forall)('forall')
            PSDeclarationRefIdListImpl(DECLARATION_REF_ID_LIST)
              PSDeclarationRefIdImpl(DECLARATION_REF_ID)
                PsiElement(PSTokenType.ID)('a')
            PsiElement(PSTokenType..)('.')
          PSDataDeclarationBodyImpl(DATA_DECLARATION_BODY)
            PsiElement(PSTokenType.{)('{')
            PSDataDeclarationTypeDefinitionListImpl(DATA_DECLARATION_TYPE_DEFINITION_LIST)
              PSDataDeclarationTypeDefinitionImpl(DATA_DECLARATION_TYPE_DEFINITION)
                PsiElement(PSTokenType.ID)('name')
                PsiElement(PSTokenType.::)('::')
                PSParenthesisedIdListImpl(PARENTHESISED_ID_LIST)
                  PSParenthesisedDeclarationRefIdListImpl(PARENTHESISED_DECLARATION_REF_ID_LIST)
                    PSDeclarationRefIdImpl(DECLARATION_REF_ID)
                      PsiElement(PSTokenType.ID)('a')
            PsiElement(PSTokenType.})('}')